### PR TITLE
Feature: Introduce Argo CD

### PR DESCRIPTION
**Pull request by werniq**
### Pull Request Description: Introduce Argo CD for Go-Gator

**Summary:**
This pull request introduces Argo CD into the project for GitOps-based continuous deployment. It includes the setup of three directories for Helm charts to manage the deployment of different components. The structure is as follows:

1. **`day-0`**: Contains essential setup for the application, including a cron job that periodically updates AWS credentials secrets. Additionally, this directory includes dependency charts for:
   - **Vertical Pod Autoscaler**: Ensures optimal resource usage by scaling pods based on usage metrics.
   - **Cert Manager**: Manages TLS certificates, ensuring secure communication.

2. **`go-gator`**: This directory contains the Helm chart for the **Go-Gator** application, the core of the project.

3. **`go-gator-operator`**: Contains the image for the **Go-Gator Operator**, which manages the lifecycle of custom resources related to **HotNews** and **Feed** CRDs.

**Key Changes:**
- Argo CD introduced for managing Kubernetes manifests and automating application delivery.

image

### CODE REVIEW

File: argo-cd/go-gator/application.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: go-gator-app
+  namespace: go-gator
+spec:
+  project: default
+  source:
```
Comment (VitaliyShevchenko) [inline argo-cd/go-gator/application.yaml:8]: are you sure this will work? where is a declaration of git project and folder which it will watch?

**Review by VitaliyShevchenko — CHANGES_REQUESTED** [review]
this should not work.

File: day-0/Chart.yaml [hunk 2]
```diff
@@ -4,3 +4,12 @@ description: A Helm chart for Kubernetes
 type: application
 version: 0.1.0
```
[the remark this answers is not in the document]
Comment (werniq) [reply day-0/Chart.yaml]: Updated version in day-0 and go-gator charts

File: argo-cd/go-gator/application.yaml (hunk 1 above)
Comment (werniq) [reply argo-cd/go-gator/application.yaml]: Created file with main application, which launches 3 apps which are in `argo-cd/apps` folder

**Pull request merged**